src/module/encounter/document.ts

```diff
@@ -92,8 +95,18 @@ class EncounterPF2e extends Combat {
             })
         );
 
-        const initiatives = rollResults.flatMap((result) =>
-            result ? { id: result.combatant.id, value: result.roll.total } : []
+        const initiatives = rollResults.flatMap((result): SetInitiativeData | never[] =>
+            result
+                ? {
+                      id: result.combatant.id,
+                      value: result.roll.total,
+                      statistic:
+                          result.roll.options.domains?.find(
+                              (s): s is SkillLongForm | "perception" =>
+                                  setHasElement(SKILL_LONG_FORMS, s) || s === "perception"
```

Comment:
Is it possible to roll initiative with lore skills? If so, I don't know if this is a safe way to find the rolled skill.

Reply:
The general rule is that lore skills can only be used for recall knowledge, but it's a special ability away from overriding that (e.g. Exploit Vulnerability--haven't surveyed for any allowing one to use a lore skill for initiative). Implementation-wise, the system only offers non-lore skills as options.

Reply:
blargh